### ISSUE TITLE
Change default of `jsx-closing-bracket-location` to `line-aligned`

### ISSUE THREAD

**Issue by golopot** [yannickcr/eslint-plugin-react#2272]
The current default of `jsx-closing-bracket-location` is `tag-aligned`. It *disallows*
```js
// line-aligned style
const Foo = () => <div 
  foo
/>
```
and allows
```jsx
// tag-aligned style
const Foo = () => <div 
  foo
                   />
```
The tag-aligned style conflicts with the rule `indent`. Also I think that `line-aligned` is a more main-stream style. The default should be changed to `line-aligned`.

**Comment by ljharb** [comment]
The indent rule needs to ignore jsx nodes to work properly with the jsx indent rules in this plugin.

Changing the default is a breaking change for minimal benefit.
